src/main/kotlin/com/dp/advancedgunnerycontrol/weaponais/tags/ConservePDAmmoTag.kt

```diff
@@ -0,0 +1,29 @@
+package com.dp.advancedgunnerycontrol.weaponais.tags
+
+import com.dp.advancedgunnerycontrol.weaponais.*
+import com.fs.starfarer.api.combat.CombatEntityAPI
+import com.fs.starfarer.api.combat.MissileAPI
+import com.fs.starfarer.api.combat.ShipAPI
+import com.fs.starfarer.api.combat.WeaponAPI
+import org.lwjgl.util.vector.Vector2f
+
+// Only fire at full ROF if target is missile or fighter and ammo < ammoThreshold
+class ConservePDAmmoTag(weapon: WeaponAPI, private val ammoThreshold: Float) : WeaponAITagBase(weapon) {
+
+    override fun isBaseAiValid(entity: CombatEntityAPI): Boolean = true
+
+    override fun computeTargetPriorityModifier(entity: CombatEntityAPI, predictedLocation: Vector2f): Float = 1f
+
+    override fun shouldFire(entity: CombatEntityAPI, predictedLocation: Vector2f): Boolean {
+        if(ammoLevel(weapon) < ammoThreshold)  {
+            return (entity as? ShipAPI)?.isFighter == true || (entity is MissileAPI) && isPD(weapon)
+        }
+        return true
+    }
+
+    override fun isBaseAiOverridable(): Boolean = true
```

Comment:
Oh right, I was worried if I set it to false then it would maybe block other tags that wanted to use customAI, just read "isBaseAIOverwritable" in TagBasedAI, and looks like it only requires one tag to be overridable. I think I can just set it to false